src/shared/mount-util.c

```diff
@@ -856,6 +948,17 @@ static int mount_in_namespace(
         if (S_ISLNK(st.st_mode)) /* This shouldn't really happen, given that we just chased the symlinks above, but let's better be safe… */
                 return log_debug_errno(SYNTHETIC_ERRNO(EOPNOTSUPP), "Source directory %s can't be a symbolic link", src);
 
+        if (!is_image && ms_nosymfollow_supported() > 0) /* Shortcut if we can use the new mount API */
```

Comment:
hmm, the propagate_path/laccess check further up really should be skipped if the new mount api exists.

i don't like overloading ms_nosymfollow_supported() for checking if the new mount api exists. please add an explicit helper, that just calls fsopen("tmpfs")